models/DataObject/ClassDefinition/Layout/Text.php

```diff
@@ -126,15 +126,15 @@ public function setBorder(bool $border): void
      */
     public function enrichLayoutDefinition($object, $context = [])
     {
-        $renderingClass = $this->getRenderingClass();
-
-        if (Tool::classExists($renderingClass)) {
-            if (method_exists($renderingClass, 'renderLayoutText')) {
-                $context['fieldname'] = $this->getName();
-                $context['layout'] = $this;
-                $result = call_user_func($renderingClass . '::renderLayoutText', $this->renderingData, $object, $context);
-                $this->html = $result;
-            }
+        $renderingClass = Model\DataObject\ClassDefinition\Helper\DynamicTextResolver::resolveRenderingClass(
+            $this->getRenderingClass()
+        );
+
+        if (method_exists($renderingClass, 'renderLayoutText')) {
```

Comment:
maybe instanceof DynamicTextLabelInterface in addition (keep method_exists for BC reasons).

Reply:
Good point, I'll add it.